src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -749,8 +833,34 @@ class NBDListIterator {
       return -ENOENT;
     }
 
+    m_mapped_info_cache.erase(pid);
+    if (!cfg->devpath.empty()) {
+      m_mapped_info_cache[pid] = *cfg;
+    }
+
     return 0;
   }
+
+  int find_proc(const std::string &devpath) {
+    namespace fs = std::experimental::filesystem;
+    for (auto &entry : fs::directory_iterator("/proc")) {
+      if (!fs::is_directory(entry.status())) {
+        continue;
+      }
+
+      int pid = atoi(entry.path().filename().c_str());
```

Comment:
Nit: you could just use `boost::lexical_cast` to determine if the directory name was an integer